src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManagerWithDataCollection.cs

```diff
@@ -131,6 +133,17 @@ public override void Close()
         {
             base.Close();
         }
+
+        /// <inheritdoc />
+        protected override TestProcessStartInfo GetTestHostProcessStartInfo(
+            IEnumerable<string> sources,
+            TestRunnerConnectionInfo connectionInfo)
+        {
+            var testHostProcessStartInfo = this.testHostManager.GetTestHostProcessStartInfo(sources, this.dataCollectionParameters.EnvironmentVariables, connectionInfo);
```

Comment:
Now we're leaking the knowledge of `ITestHostManager` to every proxy. Is it a better abstraction to just allow derived classes to update the `TestProcessStartInfo`?